### PR TITLE
support max-alive in keep-alive - #3961 

### PR DESCRIPTION
**Pull request by Kingwl**
#3961

### CODE REVIEW

**Comment by znck** [comment]
This PR is based to `dev` branch. Then why is it tagged `1.x`?

File: src/core/components/keep-alive.js [hunk 1]
```diff
@@ -47,11 +47,32 @@ export default {
...
+  methods: {
+    toCache (key: string | number, vnode: VNode) {
+      if (this.cache[key]) {
+        vnode.componentInstance = this.cache[key].componentInstance
+      } else {
+        if (this.maxAlive > 0) {
+          while (this.queue.length >= this.maxAlive) {
+            const willDestroyKey = this.queue.shift()
+            pruneCacheEntry(this.cache[willDestroyKey])
+            this.cache[willDestroyKey].data.keepAlive = false
```
Comment (Mafieu) [inline src/core/components/keep-alive.js]: Isn't that a useless line since you put everything to `null` next line ?

Comment (Kingwl) [inline src/core/components/keep-alive.js]: no it isn't
next line set a object property (a reference) only

Comment (Mafieu) [inline src/core/components/keep-alive.js]: Oh, that reference is used somewhere else then.
Thanks for the answer.

Comment (Kingwl) [inline src/core/components/keep-alive.js]: :XD

**Comment by posva** [comment]
what happened @Kingwl 🙁

**Comment by znck** [comment]
I want this too 😞

**Comment by Kingwl** [comment]
😂😂refact.....

**Pull request closed without merging**